### PR TITLE
Fixes secret recipes considering stranger reagent and inaprovaline valid.

### LINKED ISSUE #75304: secret recipes can contain unobtainable reagents
Reporting client version: 514.1589

## Round ID:
[205371](https://scrubby.melonmesa.com/round/205371)

## Testmerges:
- [JSON Logging Take Two](https://www.github.com/tgstation/tgstation/pull/73604)
- [Adds TTS to the game. Players can select their own voices in preferences.](https://www.github.com/tgstation/tgstation/pull/74775)

## Reproduction:
image
both inaprovaline and stranger reagent are not obtainable outside admin spawn

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #75304

:cl: ShizCalev
fix: Secret recipes will no longer contain inaprovaline or stranger reagent.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
@ShizCalev Strange Reagent is obtainable in-game, but the `strange_reagent/instant` isn't.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -880,6 +880,7 @@
 /datum/reagent/medicine/strange_reagent/instant
 	name = "Stranger Reagent"
 	instant = TRUE
+	chemical_flags = NONE
```
Comment (ZephyrTFA) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:883]:
```suggestion
	chemical_flags = REAGENT_NO_RANDOM_RECIPE
```

Comment (ZephyrTFA) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:883]: still needs the flag for not being allowed in random recipes

Comment (ShizCalev) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:883]: image
no you don't.

**Pull request merged**